test/service/test_tdr.py

```diff
@@ -0,0 +1,131 @@
+from functools import cached_property
+import json
+from operator import attrgetter
+from typing import (
+    Dict,
+    Iterable,
+    Mapping,
+    Union,
+)
+import unittest
+from unittest import mock
+
+from more_itertools import one
+from tinyquery import tinyquery
+
+from azul.bigquery import AbstractBigQueryAdapter
+from azul.indexer import BundleFQID
+from azul.tdr import (
+    AzulTDRClient,
+    BigQueryDataset,
+)
+from azul.types import (
+    JSON,
+    JSONs,
+)
+from azul.vendored.frozendict import frozendict
+from azul_test_case import AzulTestCase
+
+
+class TestTDRClient(AzulTestCase):
+    _gbq_adapter_mock = None
+
+    def setUp(self) -> None:
+        self.query_adapter = TinyBigQueryAdapter()
+        self._gbq_adapter_mock = mock.patch.object(AzulTDRClient,
+                                                   'big_query_adapter',
+                                                   new=property(lambda client: self.query_adapter))
```

Comment:
```suggestion
                                                   new=self.query_adapter)
```

doesn't work?

![image](https://user-images.githubusercontent.com/5143256/85889695-d1c1a280-b7a0-11ea-8f97-77bb2695c1ec.png)